microsoft-edge/progressive-web-apps-chromium/how-to/app-actions.md

```diff
@@ -0,0 +1,397 @@
+
+Example `share_target` declaration, from [Demos/wami/manifest.json](https://github.com/MicrosoftEdge/Demos/blob/main/wami/manifest.json):
+
+```json
+"share_target": {
+   "action": "./share-target",
+   "method": "POST",
+   "enctype": "multipart/form-data",
+   "params": {
+      "title": "title",
+      "text": "text",
+      "url": "url"
+   }
+}
+```
+
+See also:
+* [Receiving shared content](./share.md#receiving-shared-content)
+
+
+<!-- ====================================================================== -->
+## `launch_handler` in the web app manifest
+
+If you want to keep a single PWA window, you can set [launch_handler](https://developer.mozilla.org/docs/Web/Progressive_web_apps/Manifest/Reference/launch_handler) as `navigate-existing` in your PWA's web app manifest file (such as [Demos/wami/manifest.json](https://github.com/MicrosoftEdge/Demos/blob/main/wami/manifest.json)).
+
+Example declaration:
+
+```json
+"launch_handler": {
+   "client_mode": " navigate-existing"
+}
+```
+
+The above code is not in [Demos/wami/manifest.json](https://github.com/MicrosoftEdge/Demos/blob/main/wami/manifest.json).
+
+
+<!-- ====================================================================== -->
+## Accept entity info on Windows<!-- todo: heading + sentence 1: clarify, what is this section about?  what is "entity info"?  what is the task/accomplishment covered in this section? -->
+
+<!-- todo: To accomplish ABC, -->You can use or combine the following approaches:
+
+* Handle POST share data by using server-side code.
+   * For online users.
+
+* Use a fetch event listener to intercept the HTTP <!-- todo: POST? --> request.  This approach allows accessing the data in a service worker.
+   * This provides a better experience for offline users.
+
+To use a service worker to handle POST share data:
+<!-- todo:
+does this apply to both approaches above?
+use same wording as one of the above options.  which of the above options is this about?
+For the App Action scenario of using a service worker (approach 2?) to handle POST (approach 1?) share data, 
+-->
+* [share_target - Receiving share data using POST](https://developer.mozilla.org/docs/Web/Progressive_web_apps/Manifest/Reference/share_target#receiving_share_data_using_post)
+* [share_target - Receiving shared files](https://developer.mozilla.org/docs/Web/Progressive_web_apps/Manifest/Reference/share_target#receiving_shared_files)
+
+The following code is from the wami demo's example service worker handler, to support AI Action launch for the app.  This code is from [Demos/wami/sw.js](https://github.com/MicrosoftEdge/Demos/blob/main/wami/sw.js):
+
+```csharp
+// Extract data
+const data = {
+   title: formData.get('title') || '',
+   text: formData.get('text') || '',
+   url: formData.get('url') || ''
+};
+const files = formData.getAll('windowsActionFiles');
+```
+
+The above code uses the following `formData`:
+
+<!-- todo
+* **title:** 
```

Comment:
should `title` be mentioned in this list?  the other 3 formData lines are listed.
@novac42